### ISSUE TITLE
Opening app with Expo Go results in an unmatched route

### ISSUE THREAD

**Issue by tamagokun** [expo/router#398]
### Summary

Trying to access an app bundle through Expo Go on your device will open the app, and then try to navigate to the update bundle path, resulting in the app to show the 404 screen.

Example url: `exp://u.expo.dev/update/{uuid}/` when opening the app, will try to navigate to `exp://u.expo.dev/update/{uuid}//--/update/{uuid}`

If you create a screen to handle that route, it won't 404, and you can redirect back to `/` or do whatever you need to do as a workaround.

Is this intentional? Or is there an app.json/eas.json setting that I have missed?

### Minimal reproducible example

1. Create an app with expo-router
2. Push a bundle up to expo.dev
3. Open Expo Go
4. Navigate to the project, and the update you just pushed
5. Tap "Open"
6. See the 404. (note: this seems to NOT happen the VERY first time you open the app, but then any subsequent open causes the 404 route)

**Comment by ikapo** [comment]
Running into this issue as well

**Comment by salmoro** [comment]
Have the same issue if I'm using `initialRouteName` but will work if I have the default `app/index.js`.
Might be related to https://github.com/react-navigation/react-navigation/issues/11262

**Comment by tamagokun** [comment]
Confirmed that if you remove `initialRouteName` this doesn't happen

**Comment by mihailapuste** [comment]
I am also getting this issue. Removing `initialRouteName` does not seem to help.

**Comment by salmoro** [comment]
@mihailapuste @ikapo do you have a `index.js` /  `index.tsx` in the root folder? Also did you `expo start --clear`?

**Comment by EvanBacon** [comment]
This format is not supported at the moment (eas update in Expo Go), I’ll make a note to investigate adding support tomorrow.

**Comment by ikapo** [comment]
@salmoro Yes, and the index route renders fine. No other routes can't be navigated to though, and the site map does not show them even though I have other files in the same directory as `app/index.tsx`.

**Comment by EvanBacon** [comment]
Communicating this with the team, it's unclear if the URL format they chose has support for deep links. e.g. `exp://u.expo.dev/update/{uuid}/--/update/{uuid}` is technically correct as `/update/{uuid}` is the pathname and `/--/` is the divider rule in Expo Go. Updates URL should probably be changed to look something like: `exp://{uuid}.expo.dev/` so the pathname can be extracted correctly.

**Comment by britisharmy** [comment]
I am running into this problem too today. What exactly is the fix? Should i update anything?

**Comment by aufabdulnafea** [comment]
I am having the same problem, I just have a Drawer in my root _layout.tsx file and an index.tsx file which renders a text, I tried updating to the latest expo-router (1.5.3), but unfortunately that did not help

**Comment by meagher43** [comment]
Also having the same problem

**Comment by jordanful** [comment]
Same problem here.

**Comment by MacDev127** [comment]
Having the same problem

**Comment by veixvv** [comment]
Same problem here.

**Comment by marklawlor** [comment]
If you still having this issue, please open a new issue with a reproduction. Reply "same here" provides us with no new information, so we will not be able to fix your issue.

**Comment by tusharnagar17** [comment]
Still facing this issue on 25-Oct-2023 if there any update. Kindly replied to this thread.

**Comment by marklawlor** [comment]
@tusharnagar17 Quoting the comment directly above yours. We cannot provide an update if you do not provide more information

> If you still having this issue, please open a new issue with a reproduction. Reply "same here" provides us with no new information, so we will not be able to fix your issue.

**Comment by tusharnagar17** [comment]
1. i have create app with `npx create-expo-app@latest -e with-router`
2. create new dir named `app/` in root app folder
3. create app/_layout.js
```
import React from "react";
import { Slack } from "expo-router";

function Layout() {
  return <Slack />;
}

export default Layout;
```
4. create app/index.js
```
import { View, Text } from "react-native";

const Home = () => {
  return (
    <View>
      <Text>King is here brother Home is here!</Text>
    </View>
  );
};

export default Home;
```
4. running these command 
```
npm i -g expo-cli
expo-cli start --tunnel
```

5. now running these above command and scanning expo QR code with my android phone causing specific  error
- ERROR: Unmatched Route Page could not be found.
![image](https://github.com/expo/router/assets/75514135/c9051976-db53-42f8-9f7e-c0dc729f4ff0)

Kindly take a look at this issue community

**Comment by Tafsan-Mahmud** [comment]
> I am running into this problem too today. What exactly is the fix? Should i update anything?

just create index.tsx/index.js file in the app folder.

**Comment by Ammar-kratos** [comment]
make sure you export default component in app/index.js

**Comment by wesleyscholl** [comment]
"same here"

I solved it by creating `index.tsx` in the `/app` folder and redirecting it to `Home` component.

```
import { Redirect } from "expo-router";

export default function App() {
  return <Redirect href="/Home" />;
}

```

**Comment by laurovcn** [comment]
i'm having this problem right now, hours ago it was working well.. 

![image](https://github.com/expo/router/assets/49030558/612f456b-b873-49e9-a6d0-463fc57944a1)
